src/presentation/components/button/LogOutBtn.vue

```diff
@@ -0,0 +1,29 @@
+<template>
+<Button 
+  text="Log out"
+  :icon="IconUnlink"
+  type="primary"
+  @click="logOut"
+  />
+</template>
+
+<script setup lang="ts">
+import { IconUnlink } from '@codexteam/icons';
+import Button from './Button.vue';
+import { authService } from '@/domain/index';
```

Comment:
Accessing a domain level from Presentation level is prohibited. ESLint should warn about it, try `yarn lint`.

You can use Application Service for that reason.

I'd suggest renaming `useOAuth()` to `useAuth()` and put a logout method in there.